src/App.tsx

```diff
@@ -15,36 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = '',
+  ALPHABETICALLY = 'alphabetically',
+  LENGTH = 'length',
+}
+
+const getSortedGoods = (
+  goods: string[],
+  sortType: SortType,
+  reversed: boolean,
+): string[] => {
+  const sortedGoods = [...goods];
+
+  sortedGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABETICALLY:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortValue, setActiveButton] = useState<SortType>(SortType.DEFAULT);
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const sortGoods = (type: SortType) => {
+    setActiveButton(type);
+  };
+
+  const toggleReverse = () => {
+    setReversed(!reversed);
+  };
+
+  const resetGoods = () => {
+    setReversed(false);
+    setActiveButton(SortType.DEFAULT);
+  };
+
+  const sortedGoods = getSortedGoods(goodsFromServer, sortValue, reversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortValue !== SortType.ALPHABETICALLY,
+          })}
+          onClick={() => {
+            sortGoods(SortType.ALPHABETICALLY);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => {
+            sortGoods(SortType.LENGTH);
+          }}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortValue !== SortType.LENGTH,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={toggleReverse}
+          type="button"
+          className={cn('button is-warning', { 'is-light': !reversed })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortValue !== SortType.DEFAULT || reversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={resetGoods}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {sortedGoods.map(good => (
+          <li data-cy="Good" key={good}>
```

Comment:
[CODE KNOWLEDGE] - When rendering a list, make sure to add a unique `key` prop. Using the item name as a key is risky if the names are not guaranteed to be unique. Consider using an ID or a combination of name and index if the names may repeat.